R/plumber-step.R

```diff
@@ -307,6 +315,37 @@ PlumberEndpoint <- R6Class(
       self$path <- path
       private$regex <- createPathRegex(path, self$getFuncParams())
       path
+    },
+    #' @description convert args to specific types
+    #' @param args List of args to convert
+    convertArgs = function(args) {
+      if (!isTRUE(getOption("plumber.typedParameters", TRUE))) {
+        return(args)
+      }
+
+      params <- self$getEndpointParams()
+
+      # TODO - I suspect someone with more base-R could turn this
+      # for into some wonderful s/lapply thing...
+      converted_args <- list()
+      for (i in names(args)) {
+        arg <- args[[i]]
+        param <- params[[i]]
+        converter <- identity
+        if (!is.null(param)) {
+          # NOTE: areArrays is FALSE here as we do not want to parse comma
+          #    separated strings (or do we?)
+          converter <- typesToConverters(param$type, areArrays = FALSE)
+          if (is.null(converter)) {
+            converter <- identity
+          } else {
+            converter <- converter[[1]]
+          }
+        }
+        converted_args[[i]] <- converter(arg)
```

Comment:
if we had access to a `can_convert` as well as a `converter` here then we could trigger a 400 from here... although I'm not entirely sure how I'd trigger a response from this function - I guess we could return some kind of union of *args plus errors* from this function and then `plumberStep$exec` could return a 400 error `res` instead of calling the actual endpoint handler? I might need a little nudge in the right direction on that code (or someone else can code it)